### PR TITLE
fix(tile): improve interactive element detection to include aria-role

### LINKED ISSUE #13759: [a11y]: IBM Accessibility checker reports error on Expandable Tile with Interactive
### Package

@carbon/react

### Browser

_No response_

### Operating System

_No response_

### Package version

1.23.0

### React version

16.4

### Automated testing tool and ruleset

IBM Equal Accessibility Checker with latest ruleset

### Assistive technology

_No response_

### Description

There are 2 issues:

1.  I have a StructuredList inside the Expandable tile.  It doesn't have interactive content so the tile gets rendered as a button.  However, the StructuredList elements have roles and buttons can't have children with roles.

So, to work around this problem, I added a button with display:none in order to force the expandable tile to render the version with interactive content, making the button just the expando icon instead of the whole tile.

Now I run into the issue, reproducible in the storybook, that the tile can't have attribute aria-expanded="false"
image

In addition to fixing the aria-expanded issue, it would be great if I could get the "interactive" version without having to put a fake interactive element in my code.

### WCAG 2.1 Violation

_No response_

### Reproduction/example

see story book

### Steps to reproduce

see storybook screen shot in description

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by tay1orjones**
Closes #13759 

#### Changelog

**New**

- add new `getRoleContent()` for checking if elements within a tree have an aria-role assigned to them

**Changed**

- use `getRoleContent()` in tile to render the 'expandable with interactive' tile when children elements have an aria-role assigned
- add tests to cover this case

#### Testing / Reviewing

- If an element is rendered within the tile that contains a role but is not interactive, the 'expandable with interactive' tile should be rendered
- Existing tile stories should operate the same as before.
- The existing tile stories in the feature flags section should also operate the same as before.

### CODE REVIEW

File: packages/react/src/components/Tile/Tile.tsx [hunk 1]
```diff
@@ -663,10 +666,14 @@ export const ExpandableTile = React.forwardRef<
       return;
     }
 
-    if (!getInteractiveContent(belowTheFold.current)) {
-      setInteractive(false);
-      return;
-    } else if (!getInteractiveContent(aboveTheFold.current)) {
```
Comment (tay1orjones) [inline packages/react/src/components/Tile/Tile.tsx:669]: I also found a bug here where the above-the-fold content was not being checked for interactive elements. This was originally introduced with https://github.com/carbon-design-system/carbon/pull/13365. 

When `interactive` was inversed to be true by default we no longer needed the optimization to bail out of checking the below-the-fold content for interactive elements if the above-the-fold did have interactive elements (line 668). 

The new approach ensures that both the above and below content are checked - if either has interactive elements or a role, the 'expandable with interactive' tile is rendered.

**Review by guidari — COMMENTED** [review]
Looks good to me the changes!! 🚀

Just one thing I noticed is about the a11y violation.

Screenshot 2023-07-20 at 13 26 15

If we add a role to the `ExpandableTile` the violation is gone.

Screenshot 2023-07-20 at 13 25 03

**Comment by tay1orjones** [comment]
@guidari thanks for catching that! I got so caught up in getting it to detect children with a `role` that I forgot to address the root issue of the a11y violation 🤦 

I added `aria-owns` to the button controlling the belowTheFold content which solves the violation.

**Comment by tay1orjones** [comment]
@guidari @alisonjoseph This one's ready for review 👍

**Review by guidari — APPROVED** [review]
LGTM! 🚀

**Review by alisonjoseph — APPROVED** [review]
LGTM!

**Pull request merged**